### ISSUE TITLE
Maven release plugin 3.x considerations due to shallow checkout

### ISSUE THREAD

**Issue by hazendaz** [mathieucarbou/license-maven-plugin#206]
Hi @mathieucarbou ,

Maven release plugin 3.0.0-m1 changed to shallow clones [here](https://issues.apache.org/jira/browse/MRELEASE-993).  There were some bugs with 3.0.0-m1 so I had not noticed the situation until now with 3.0.0-m4.  Basically during the 'release:perform', this plugin will fail on every call as it cannot determine the commit.  I confirmed the issue is present with both 3.0.0-m1 and 3.0.0-m4.  I think documentation should be added to note that license plugin should just be skipped during release cycle.  Thoughts?

**Comment by mathieucarbou** [comment]
Hi @hazendaz !
What you do mean by _it cannot determine the commit_ ? You are talking about the optional MLP git extension I guess ?

**Comment by hazendaz** [comment]
@mathieucarbou Yes, sorry just that situation.  When it tries to get the timestamps using git extension (jgit).

**Comment by mathieucarbou** [comment]
What I will say could still be subject to an argumentation with pros and cons, but I think that this change of (global) behaviour might also impact other plugins relying on git history before. I would find strange also that Maven forces all plugin developers to update their plugin because of a new clone strategy.

IMO, [the way they have coded the change is completely stupid and rackless](https://github.com/apache/maven-release/commit/39708b03374b007df5fc7b6f2f7943693886af52#diff-da10825f898b1702e8d0e8676cd9ee709786f82879de37ce90e318ed7594b043R215-R216). Any big behavioural change like, if not done in a backward compatible way, should at least provide an option to switch to the old clone behaviour.

I would perhaps instead provide some workarounds, like:

- skipping the license for the release:perform goal `-Darguments="-Dlicense.skip=true`
- using the `<releaseProfiles>` feature
- using custom profiles and / or properties

So in conclusion, yes I agree with your conclusion 🤣 

A little doc update would be helpful indeed.

**Comment by rnc** [comment]
@mathieucarbou I went down the skip license route with `releaseProfiles` route (see above linked PR)
